token/cli/src/main.rs

```diff
@@ -482,6 +483,94 @@ fn command_unwrap(config: &Config, address: Pubkey) -> CommandResult {
     Ok(Some(transaction))
 }
 
+fn command_approve(
+    config: &Config,
+    account: Pubkey,
+    ui_amount: f64,
+    delegate: Pubkey,
+) -> CommandResult {
+    println!(
+        "Approve {} tokens\n  Account: {}\n  Delegate: {}",
+        ui_amount, account, delegate
+    );
+
+    let source_account_balance = config
+        .rpc_client
+        .get_token_account_balance_with_commitment(&account, config.commitment_config)?
+        .value;
+    let source_account = config
+        .rpc_client
+        .get_account_with_commitment(&account, config.commitment_config)?
+        .value
+        .unwrap_or_default();
+    let data = source_account.data.to_vec();
+    let mint_pubkey = Account::unpack_from_slice(&data)?.mint;
```

Comment:
This can be replaced by one call to `get_token_account_with_commitment()`

Reply:
Oh, totally! Thanks for the reminder. I'll fixup `transfer` while I'm on that job.